larray/inout/array.py

```diff
@@ -497,21 +495,33 @@ def read_hdf(filepath_or_buffer, key, fill_value=np.nan, na=np.nan, sort_rows=Fa
                       FutureWarning, stacklevel=2)
 
     key = _translate_key_hdf(key)
-    pd_obj = pd.read_hdf(filepath_or_buffer, key, **kwargs)
-    if type == 'LArray':
-        return df_aslarray(pd_obj, sort_rows=sort_rows, sort_columns=sort_columns, fill_value=fill_value, parse_header=False)
-    elif type == 'Axis':
-        if name is None:
-            name = str(pd_obj.name)
-        return Axis(labels=pd_obj.values, name=name)
-    elif type == 'Group':
-        _name, axis_key = str(pd_obj.name).split('@')
-        if name is None:
-            name = _name
-        axis = read_hdf(filepath_or_buffer, axis_key, type='Axis')
-        return LGroup(key=pd_obj.values, name=name, axis=axis)
-    else:
-        return None
+    res = None
+    with pd.HDFStore(filepath_or_buffer, **kwargs) as store:
+        pd_obj = store.get(key)
+        attrs = store.get_storer(key).attrs
+        # for backward compatibility but any object read from an hdf file should have an attribute 'type'
+        _type = attrs.type if 'type' in dir(attrs) else 'LArray'
+        if _type == 'LArray':
+            res = df_aslarray(pd_obj, sort_rows=sort_rows, sort_columns=sort_columns, fill_value=fill_value, parse_header=False)
+        elif _type == 'Axis':
```

Comment:
It kinda bothers me that this code is in the inout/**array** module. Not urgent to fix this but still... Here are a few potential solutions:
* defer to some function in a inout/axis module when type == 'Axis'
* rename inout/array to inout/core
* have inout/api with read_hdf which defers to functions in inout/array and inout/axis modules depending on type.

Reply:
Let's start with option 2 (rename inout/array to inout/core). It will be easy to move from option 2 to option 1 or 3 later if we changed our mind.